images/pstree.proto

```diff
@@ -6,4 +6,11 @@ message pstree_entry {
 	required uint32			pgid		= 3;
 	required uint32			sid		= 4;
 	repeated uint32			threads		= 5;
+/*
+ * These fields are reserved for Virtuozzo criu nested pidns feature
+ *	repeated uint32			ns_pid		= 6;
+ *	repeated uint32			ns_pgid		= 7;
+ *	repeated uint32			ns_sid		= 8;
+ *	repeated ns_tid			tids		= 9;
+ */
```

Comment:
I don't mind, but still wondering - can you use in vz-criu some upper number instead?
Like:
```
 	repeated uint32			ns_pid		= 1021;
 	repeated uint32			ns_pgid		= 1022;
 	repeated uint32			ns_sid		= 1023;
 	repeated ns_tid			tids		= 1024;
```
That way you wouldn't need to reserve any fields..

Reply:
+1

Reply:
If we port one vzcriu feature to mainstream (e.g. multiple unix socket bindmounts) we will have to port those "upper" number ids to mainstream too (so that vzcriu can be re-based on top).

If such a big numbers are fine in mainstream criu and nobody exept Virtuozzo will use them we can follow this convention.

Reply:
Using higher numbers for vzcriu features sounds indeed like a good idea.

> If we port one vzcriu feature to mainstream (e.g. multiple unix socket bindmounts) we will have to port those "upper" number ids to mainstream too (so that vzcriu can be re-based on top).

I do not think this is a big problem. The protobuf numbers do not have to consecutive. We should probably have somewhere a hint why there are high protobuf field numbers.

Reply:
We doing all possible to port all features from vzcriu to main the main CRIU project. The idea to make all Virtuozzo-specific image fields numbers start from `1000` is good as for me. But still remains the problem with names of fields in protobuf. Because we already meet name collisions in the past (`mnt_id` in `sk-unix.proto` as example). And I think that even after we start numbering from `1000` in any way we will have to prepare reservations for new field names. Is it OK?